containers/Investment/RevolvingPool/View/EpochOverview.tsx

```diff
@@ -25,31 +43,145 @@ const EpochOverview: React.FC<Props> = (props: Props) => {
 
+  const [timePassed, setTimePassed] = React.useState(0)
+
+  useInterval(() => {
+    setTimePassed(new Date().getTime() / 1000 - props.epochData.lastEpochClosed)
+  }, 1000)
+
+  // const totalPendingInvestments =
+  //   pool.data && (pool.data as PoolDataV3).senior
+  //     ? (pool.data as PoolDataV3).junior?.pendingInvestments!.add((pool.data as PoolDataV3).senior?.pendingInvestments!)
+  //     : new BN(0)
+
   return (
-    <Box width="420px" pad="medium" margin={{ bottom: 'medium' }}>
-      <Box direction="row" margin={{ top: '0', bottom: 'small' }}>
-        <Heading level="5" margin={'0'}>
-          Current Epoch
-        </Heading>
-      </Box>
+    <Box direction="column">
+      <Box width="420px" margin={{ top: 'small', bottom: 'medium' }}>
+        <Box direction="row" margin={{ top: '0', bottom: 'small' }}>
+          <Heading level="5" margin={'0'}>
+            Current Epoch
+          </Heading>
+        </Box>
+
+        <Table>
+          <TableBody>
+            <TableRow>
+              <TableCell scope="row">Epoch #</TableCell>
+              <TableCell style={{ textAlign: 'end' }}>{props.epochData.id}</TableCell>
+            </TableRow>
+            <TableRow>
+              <TableCell scope="row">Time passed since start of current epoch</TableCell>
+              <TableCell style={{ textAlign: 'end' }}>{secondsToHms(timePassed)}</TableCell>
+            </TableRow>
+            <TableRow>
+              <TableCell scope="row">Minimum epoch duration</TableCell>
+              <TableCell style={{ textAlign: 'end' }}>{secondsToHms(props.epochData.minimumEpochTime)}</TableCell>
+            </TableRow>
+            <TableRow>
+              <TableCell scope="row">Pool Reserve current</TableCell>
+              <TableCell style={{ textAlign: 'end' }}>
+                {pool.data &&
+                  addThousandsSeparators(toPrecision(baseToDisplay((pool.data as PoolDataV3).reserve, 18), 2))}{' '}
+                DAI
+              </TableCell>
+            </TableRow>
+          </TableBody>
+        </Table>
 
-      <Table>
-        <TableBody>
-          <TableRow>
-            <TableCell scope="row">Epoch #</TableCell>
-            <TableCell style={{ textAlign: 'end' }}>{props.epochData.id}</TableCell>
-          </TableRow>
-        </TableBody>
-      </Table>
-
-      <Box gap="small" justify="end" direction="row" margin={{ top: 'small' }}>
-        {props.epochData.state === 'can-be-closed' && (
-          <Button label="Close epoch" onClick={solve} disabled={disabled} />
-        )}
-        {props.epochData.state === 'challenge-period-ended' && (
-          <Button label="Execute orders" onClick={execute} disabled={disabled} />
-        )}
+        <Box gap="small" justify="end" direction="row" margin={{ top: 'small' }}>
+          {props.epochData.state === 'can-be-closed' && (
+            <Button label="Close epoch" primary onClick={solve} disabled={disabled} />
+          )}
```

Comment:
Shall we disable this generally until it works?

Reply:
It works already, as this is also the button to close the epoch. It just fails if the epoch cannot be immediately executed, because then the solver needs to be run.

Reply:
Got it, thanks!